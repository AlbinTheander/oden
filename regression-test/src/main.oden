;; -*- mode: lisp;-*-

;; package declaration
(pkg main)

;; import declaration
(import fmt)

;; function definition can also use the usual scheme shorthand
(def (identity-string x) (++ "" x))

;; value definition
(def result (++ (identity-string "Hello") ", world!"))

;; main function definition, must have type (-> unit)
(def main (fn () (fmt.Println result)))
