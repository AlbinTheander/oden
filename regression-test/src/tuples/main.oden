package tuples/main

import fmt

nothing = ()

stuff :: (int, (), string)
stuff = (1, nothing, "yey")

main :: -> ()
main -> {
  fmt.Println(stuff)

  if (1, 2, 3) == (1, 2, 3) then {
    fmt.Println("Equality works.")
  } else {
    fmt.Println("OMG no!")
  }

  nothing
}
