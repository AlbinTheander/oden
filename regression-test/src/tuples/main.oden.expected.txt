{1 {} yey}
Equality works.
